### PR TITLE
fix: Validation error when creating preexisting username revised

### PR DESCRIPTION
**Pull request by techeclecticdesign**
Thank you everyone for your support with this commit.
Also thank you Preston, for showing me that all I had to do was use 127.0.0.1 instead of localhost.  I can't believe how easy that was.

### CODE REVIEW

**Review by calisio — APPROVED** [review]
This looks good to me. Will wait for one more approval from @PThorpe92 or @jtucholski. Also, before we merge, would you be able to amend your commit message for it to pass the conventional commit?

File: backend/src/database/users.go
Comment (jtucholski) [inline backend/src/database/users.go:170]: Any chance we can rename this? IsValidUsername implies a check to make sure the name doesn't violate any rules (eg. no spaces) whereas this function checks to see if the username has already been taken.

Comment (techeclecticdesign) [reply backend/src/database/users.go:170]: Good catch I wasn't using my head there.

**Comment by jtucholski** [comment]
Great, appreciate all of the changes!

**Review by jtucholski — APPROVED** [review]

**Comment by jtucholski** [comment]
This should close Issue #323 when it's merged.

**Pull request merged**
